airsonic-main/src/main/java/org/airsonic/player/spring/ServletConfiguration.java

```diff
@@ -0,0 +1,63 @@
+package org.airsonic.player.spring;
+
+import org.airsonic.player.controller.PodcastController;
+import org.airsonic.player.service.SecurityService;
+import org.airsonic.player.service.SettingsService;
+import org.airsonic.player.theme.CustomThemeResolver;
+import org.airsonic.player.theme.CustomThemeSource;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.support.ResourceBundleMessageSource;
+import org.springframework.web.servlet.ThemeResolver;
+import org.springframework.web.servlet.ViewResolver;
+import org.springframework.web.servlet.handler.SimpleUrlHandlerMapping;
+import org.springframework.web.servlet.view.InternalResourceViewResolver;
+import org.springframework.web.servlet.view.JstlView;
+
+import java.util.Properties;
+
+@Configuration
+public class ServletConfiguration {
+
+    @Bean
+    public SimpleUrlHandlerMapping urlMapping(PodcastController podcastController) {
+        SimpleUrlHandlerMapping handlerMapping = new SimpleUrlHandlerMapping();
+        handlerMapping.setAlwaysUseFullPath(true);
+        Properties properties = new Properties();
+        properties.put("/podcast/**", podcastController);
+        handlerMapping.setMappings(properties);
```

Comment:
set a mapping handling order here please. Otherwise you're leaving it to chance and the ResourceHandler might usurp the request with any changes or upgrades.
```
// Default is intmax/lowestprecedence, so need to set a higher priority than
// ResourceHttpRequestHandler (which is intmax-1). Otherwise, that will intercept every request before it gets here
  handlerMapping.setOrder(Integer.MAX_VALUE - 2);
```

Also consider renaming it to podcastUrlMapping, given that is literally the only thing it maps (everything maps to jsps or resources)?

Reply:
Changed the name. I'm leaving the order unspecified like it was in the previous xml config. A separate PR can change it.